rwengine/src/ai/CharacterController.cpp

```diff
@@ -209,8 +214,8 @@ bool CharacterController::checkForObstacles()
     for (const auto &obj : character->engine->pedestrianPool.objects) {
         // Verify that the character isn't the driver and is walking
         if (obj.second.get() != character &&
-            static_cast<CharacterObject *>(obj.second.get())->getCurrentVehicle() ==
-                nullptr) {
+            dynamic_cast<CharacterObject *>(obj.second.get())
```

Comment:
In my opinion dynamic_cast is unneeded, aka it'll only introduce extra overhead. Another way to prevent this is activate undefined behavior sanitizer on CI.